plans/issue-2139-pending-messages-migration.md

```
@@ -0,0 +1,247 @@
+This places the call in the same relative position as `rebuildPendingMessagesForSelfHealingClaim` in `runner.ts:41`.
+
+### 1B. Add the method body
+
+Insert immediately before `addObservationsUniqueContentHashIndex` (around SessionStore.ts:1041), following the docstring pattern of the two adjacent mirrors:
+
+```ts
+/**
+ * Add tool_use_id and worker_pid columns + indexes to pending_messages (migration 28).
+ *
+ * Mirrors MigrationRunner.rebuildPendingMessagesForSelfHealingClaim so bundled
+ * artifacts that embed SessionStore (e.g. worker-service.cjs, context-generator.cjs)
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)